workorders.lic

```diff
@@ -332,51 +332,80 @@ class WorkOrders
     end
   end
 
-  def count_combine_rem(stock_room, herb, herb_stock, quantity, herb2)
-    if inside?("first #{herb}", @bag) && inside?("second #{herb}", @bag)
-      bput("get #{herb} from my #{@bag}", 'You get')
-      bput("get #{herb} from my #{@bag}", 'You get')
-      case bput('combine', 'You combine', 'That stack of herbs')
-      when 'That stack of herbs'
-        stow_tool(left_hand)
-        stow_tool(right_hand)
-      when 'You combine'
-        case bput("get #{herb} from my #{@bag}", 'You get', 'What were')
-        when 'You get'
-          case bput('combine', 'You combine', 'That stack of herbs')
-          when 'That stack of herbs', 'You combine'
-            stow_tool(left_hand)
-            stow_tool(right_hand)
-          end
+  # Function which will attempt to combine any stacks of herbs in your inventory, will count them, and determine if more need to be ordered.
+  def count_combine_rem(stock_room, quantity, herb, herb_stock)
+  # Initialize variables
+    found_stack = true;
+    herb_volume_total = 0;
+    last_herb_volume = 0;
+    last_descriptor = "";
+    stack_descriptor = "";
+
+    # We have to only ever use the last word in a multi-word herb.  We will need to be careful when counting because of this.
+    # This is to resolve "red flower" vs "blue flower" because "count third red flower in my bag" fails.
+    # We must instead do a generic "tap third flower in my bag" and determine if we're tapping the right kind of flowers from the response, then actually count them.
+    herb_for_tapping = (herb.gsub(/\s+/m, ' ').strip.split(" ")).last
+
+    # Calculate the volume of the herbs that we will be searching our inventory for and/or buying.
+    # Assume that each unit we need to craft will take 25 volumes of our herb.
+    # Also assume that we also always need double the supply that the workorder would normally take in the event that some of our product is not high enough quality.
+    need_herb_volume = ((quantity * 25.0) * 2)
+
+    # Loop through all the stacks of herbs we might have in our backpack and find out how much volume we have total.
+    descriptors = [ "first", "second", "third", "fourth", "fifth", "sixth", "seventh", "eighth", "ninth", "tenth", "eleventh", "twelfth" ]
+    begin
+      stack_descriptor = descriptors.shift
+
+      # TODO: Solve the Inkin problem, which (very rarely) results in never counting any herbs, and always buying extra.
+      # The "You lightly tap" result here is what happens if you try to do this in a room with somebody having a name that start with "In" .... like... "Inkin"
+      # example:
+      #   [workorders]>tap first flower in my haversack
+      #   You lightly tap Inkin on the shoulder.
+      /You tap (.*) inside your|I could not find|You lightly tap/ =~ bput("tap #{stack_descriptor} #{herb_for_tapping} in my #{@bag}", 'You tap (.*) inside your', 'I could not find', 'You lightly tap')
```

Comment:
That seems like the right choice, but unfortunately it does not work here.
If you look at the logic, we are tricking DR in to giving us information we didn't ask for.  This is not a simple "see if I have this thing in this bag"

Detail:
`> tap third flower in my haversack`

Note that we are using the generic noun here.  This is because if we try to `>tap third blue flower in my haversack` it has no idea what we are doing.
However.

The response to:
`> tap third flower in my haversack`
is:
`You tap some dried red flowers inside your spidersilk haversack.`

DR has given us additional information back about what the type of flowers we just tapped are, and we need to have that text available for regular expression logic.

This is the engine by which we are able to separate and count red flowers differently than blue flowers.  It's basically a workaround because DR is stupid.